lib_test/unit_tests.ml

```diff
@@ -264,6 +262,46 @@ let test_ca_325844 () =
   let rrd = Rrd_unix.of_file (Filename.concat "test_data"  "flip_flop.xml") in
   test_ranges rrd ()
 
+let suite_create_multi =
+  let module RU = Rrd_updates in
+
+  let assert_size t =
+    (** we can't to check that the number of rows is consistent,
+        since this is defined purely by the number of rows
+      * the number of columns should match the number of items in the legend -
+        each element in the legend array defines the contents for one column *)
+    let num_cols_in_legend = Array.length t.RU.legend in
+    t.RU.data |> Array.iteri (fun i r ->
+      Alcotest.(check int)
+        (Printf.sprintf "number of cols in legend must matche number of cols in row[%i]" i)
+        num_cols_in_legend
+        (Array.length r.RU.row_data)
+    )
+  in
+  let test_no_rrds () =
+    Alcotest.check_raises "should raise error" (Failure "hd") (fun () -> let _ = RU.create_multi [] 0L 1L None in ())
+  in
+  (** confusingly, rows in an rra are used to define the cols in the rrd_updates/ xml...
+    * essentially we usually expect 'rows' in each rrd to be the same (test_rows_with_same_num_cols)
+    * however, we should also handle the case where they are not (test_rows_with_different_num_cols) *)
+  let matrix =
```

Comment:
Don't just name it matrix please, `valid_rrd_list` or something similar goes a long way when reading the tests.